### PR TITLE
{AKS}Replace functions in aks-preview with the ones in azure-cli

### PR DESCRIPTION
**Pull request by MartinForReal**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Review by FumingZhang — COMMENTED** [review]
Currently [aks-preview](https://github.com/Azure/azure-cli-extensions/blob/main/src/aks-preview/azext_aks_preview/azext_metadata.json) depends on [azure-cli 2.44.0](https://github.com/Azure/azure-cli/tree/azure-cli-2.44.0/src/azure-cli/azure/cli/command_modules/acs). Could you please help confirm the imports already exists in this version?

**Comment by MartinForReal** [comment]
These functions has been merged in previous versions. Or the test will fail.

**Review by FumingZhang — APPROVED** [review]
LGTM

File: src/aks-preview/HISTORY.rst [hunk 1]
```diff
@@ -11,6 +11,7 @@ To release a new version, please select a new version number (usually plus 1 to
 
 Pending
 ++++++
+* move loadbalancer/natgateway util functions to azure-cli and update reference in aks-preview project.
```
Comment (zhoxing-ms) [inline src/aks-preview/HISTORY.rst:14]: Please update the `azext.minCliCoreVersion` in `azext_metadata.json`

Comment (MartinForReal) [reply src/aks-preview/HISTORY.rst:14]: bumped azure-cli to 2.49

**Comment by zhoxing-ms** [comment]
@MartinForReal Could you please resolve these conflicts and CI issues?

File: src/aks-preview/README.rst [hunk 2]
```diff
@@ -39,5 +39,7 @@ Dependency between aks-preview and azure-cli/acs (azure-cli-core)
       - >= `\2.38.0 <https://github.com/Azure/azure-cli/releases/tag/azure-cli-2.38.0>`_, 2022/07/05
     * - 0.5.119 ~ 0.5.124
       - >= `\2.43.0 <https://github.com/Azure/azure-cli/releases/tag/azure-cli-2.43.0>`_, 2022/12/06
-    * - 0.5.125 ~ latest
-      - >= `\2.44.0 <https://github.com/Azure/azure-cli/releases/tag/azure-cli-2.44.0>`_, 2023/01/10
\ No newline at end of file
+    * - 0.5.125 ~ 0.5.150
+      - >= `\2.44.0 <https://github.com/Azure/azure-cli/releases/tag/azure-cli-2.44.0>`_, 2023/01/10
+    * - 0.5.152 ~ latest
+      - >= `\2.49.0 <https://github.com/Azure/azure-cli/releases/tag/azure-cli-2.44.0>`_, 2023/08/11
```
Comment (FumingZhang) [inline src/aks-preview/README.rst:45]:
```suggestion
      - >= `\2.49.0 <https://github.com/Azure/azure-cli/releases/tag/azure-cli-2.49.0>`_, 2023/05/23
```
date here is the release date for azure-cli

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 3]
```diff
@@ -3697,6 +3697,10 @@ def update_mc_profile_preview(self) -> ManagedCluster:
         mc = self.update_network_plugin_settings(mc)
         # update outbound type
         mc = self.update_outbound_type_in_network_profile(mc)
+        # update loadbalancer profile
+        mc = self.update_load_balancer_profile(mc)
+        # update natgateway profile
+        mc = self.update_nat_gateway_profile(mc)
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:3703]: No need to call these functions again, they've been called in `self.update_mc_profile_default() `

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 4]
```diff
@@ -3128,6 +3095,39 @@ def update_load_balancer_profile(self, mc: ManagedCluster) -> ManagedCluster:
             )
         return mc
 
+    def update_outbound_type_in_network_profile(self, mc: ManagedCluster) -> ManagedCluster:
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:3098]: If the implementation here is exactly the same as in azure-cli, may omit the func

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 5]
```diff
@@ -3128,6 +3095,39 @@ def update_load_balancer_profile(self, mc: ManagedCluster) -> ManagedCluster:
...
+        :return: the ManagedCluster object
+        """
+        self._ensure_mc(mc)
+
+        outboundType = self.context.get_outbound_type()
+        if outboundType:
+            mc.network_profile.outbound_type = outboundType
+        return mc
+
+    def update_nat_gateway_profile(self, mc: ManagedCluster) -> ManagedCluster:
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:3109]: If the implementation here is exactly the same as in azure-cli, may omit the func

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 4 above)
Comment (MartinForReal) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:3098]: This function is not introduced in azure-cli 2.49

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 3 above)
Comment (MartinForReal) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:3703]: the implementation is not added in azure-cli 2.49

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 5 above)
Comment (MartinForReal) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py]: removed. Thanks!

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 3 above)
Comment (MartinForReal) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py]: Removed. Thanks!

File: src/aks-preview/HISTORY.rst [hunk 6]
```diff
@@ -9,8 +9,10 @@ If there is no rush to release a new version, please just add a description of t
 
 To release a new version, please select a new version number (usually plus 1 to last patch version, X.Y.Z -> Major.Minor.Patch, more details in `\doc <https://semver.org/>`_), and then add a new section named as the new version number in this file, the content should include the new modifications and everything from the *Pending* section. Finally, update the `VERSION` variable in `setup.py` with this new version number.
 
-Pending
+0.5.152
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:12]: May keep Pending

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**